src/SSHDebugPS/ConnectionManager.cs

```diff
@@ -1,31 +1,68 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Microsoft.SSHDebugPS.VS;
-using Microsoft.VisualStudio.Shell;
-using Microsoft.VisualStudio.Shell.Interop;
 using System;
 using System.Globalization;
 using System.Linq;
 using liblinux;
 using liblinux.Persistence;
-using Microsoft.DebugEngineHost;
-using Task = System.Threading.Tasks.Task;
+using Microsoft.SSHDebugPS.Docker;
+using Microsoft.SSHDebugPS.SSH;
+using Microsoft.SSHDebugPS.VS;
 using Microsoft.VisualStudio.Linux.ConnectionManager;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
 
 namespace Microsoft.SSHDebugPS
 {
     internal class ConnectionManager
     {
-        internal static Connection GetInstance(string name, ConnectionReason reason)
+        public static Connection GetDockerConnection(string name)
+        {
+            Connection remoteConnection = null;
+            DockerTransportSettings settings = null;
+            // Assume format is <server>/<container> where if <server> is specified, it is for SSH
+            string[] connectionStrings = name.Split('/');
+
+            string displayName;
+            string containerName;
+            if (connectionStrings.Length == 1)
+            {
+                // local connection
+                containerName = connectionStrings[0];
+                settings = new DockerExecShellSettings(containerName, isUnix: false);
```

Comment:
>isUnix [](start = 70, length = 6)

I am assuming this is supposed to indicate if the host is unix or Windows? If so, I might suggest naming this `isHostUnix`. If not, this is wrong - Windows can host Linux containers. I would assume it is the most common option.

Reply:
I missed one. I had the other places where it was renamed. I'll fix it.

---
In reply to: [253704386](https://github.com/Microsoft/MIEngine/pull/831#discussion_r253704386) [](ancestors = 253704386)